pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -418,19 +418,34 @@ func parseStaticRoute(route *zconfig.IPRoute, config *types.NetworkInstanceConfi
 	if err != nil {
 		return fmt.Errorf("destination network is invalid: %w", err)
 	}
-	if route.Gateway == "" {
-		return errors.New("missing gateway IP address")
+	if route.Gateway == "" && route.Port == "" {
+		return errors.New("missing both gateway IP address and port label")
 	}
-	gatewayIP := net.ParseIP(route.Gateway)
-	if gatewayIP == nil {
-		return errors.New("gateway IP address is invalid")
+	var gatewayIP net.IP
+	if route.Gateway != "" {
+		gatewayIP = net.ParseIP(route.Gateway)
+		if gatewayIP == nil {
+			return errors.New("gateway IP address is invalid")
+		}
+		if gatewayIP.IsUnspecified() {
+			return errors.New("gateway IP address is all-zeroes")
+		}
 	}
-	if gatewayIP.IsUnspecified() {
-		return errors.New("gateway IP address is all-zeroes")
+	customProbe, err := parseConnectivityProbe(route.GetProbe().GetCustomProbe())
+	if err != nil {
+		return fmt.Errorf("invalid connectivity probe config: %v", err)
 	}
 	config.StaticRoutes = append(config.StaticRoutes, types.IPRouteConfig{
-		DstNetwork: dstNetwork,
-		Gateway:    gatewayIP,
+		DstNetwork:      dstNetwork,
+		Gateway:         gatewayIP,
+		OutputPortLabel: route.Port,
```

Comment:
Does it make sense to validate route.Port here (just like we validate aka parse the gateway) before using it? Or have zedrouter return an error should it not find the port?

Reply:
There is a mechanism in zedrouter to mark the NI with an error when a route has port label configured but it doesn't match any port (with the currently applied DPC). Still the NI is activated, just the problematic route(s) is/are skipped.
When DPC is changed and route is suddenly matching at least one port, the error is cleared and the route is added into the routing table.
The reason why we handle this differently to e.g. gateway IP validation, is that port missing can be temporary issue (e.g. due to a DPC fallback) and the error can be later cleared.

Reply:
Ok